modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ConfigurationStateTracker.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.manager;
+
+import static org.mule.util.Preconditions.checkState;
+import org.mule.extension.introspection.Configuration;
+import org.mule.extension.introspection.Extension;
+import org.mule.extension.introspection.Operation;
+import org.mule.extension.runtime.ConfigurationInstanceProvider;
+import org.mule.extension.runtime.OperationContext;
+import org.mule.extension.runtime.OperationExecutor;
+import org.mule.util.CollectionUtils;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.commons.collections.Predicate;
+
+/**
+ * A utility class to track state related to registered instances
+ * which are realisations of {@link Configuration} models owned
```

Comment:
because they are not instances of the Configuration interface, their objects of unknown types which are consistent with a Configuration model